### ISSUE TITLE
Unauthorized Issues

### ISSUE THREAD

**Issue by eaba** [IdentityServer/Thinktecture.IdentityServer3#898]
Hello, I am quite new to Thinktecture Identity Server.
Am experiencing Unauthorized Issues when calling Web Api and MVC controller.
I have followed the examples, am getting the token from the Identity Server; 
Please is there something am missing? I need to setup it up reliably without trial and errors in case of next time i experience such issue.

Kindly Help me out.

Can we also get WCF Sample also?

Thanks

**Comment by brockallen** [comment]
You need to do more investigation into the problem. I'd suggest enabling logging and see what the logs tell you.

Also, we don't support WCF.

**Comment by eaba** [comment]
Is it log at the Identity server end? If yes,I have done that and I see that there is not communication between the Web Api and The Idsrv. 
How can I enable log for my Web Api?

**Comment by eaba** [comment]
IN OTHER WORDS: WHEN THE USER IS NOT AUTHENTICATED, RESOURCE SERVER IS NOT REDIRECTING TO THE IDENTITY SERVER LOGIN PAGE BUT IT INSTEAD DISPLAYS "HTTP Error 403.14 - Forbidden". Which is not the expected behavior. How can i fix this?
Thanks

**Comment by brockallen** [comment]
403 means you have a valid token but simply aren't allowed to access the app. What does the `WWW-Authenticate` response header say?

**Comment by eaba** [comment]
Apologies! Sorry i get that error when i launch the server website.
Please this is my setup:
my application is in layers each layer is secured
the first layer users login to obtain token, make an ajax call to the second which makes a client credential request to obtain a token and  calls third layer using httpclient with the bearer set to the token it obtained. I dont get unauthorized issue with the first and second layer. The second layer fails authenticated call to the third layer. The third layer is not communicating with the identity server maybe as a rest of poor configuration due to conflicting nuget packages. I dont know what i did to resolve this issue when the first layer was acting up. Now i want to know for sure what the problem have been. I need your help. Thanks

**Comment by brockallen** [comment]
As I said, look at the 403 HTTP response and see what the WWW-Authenticate header says. Another thing to check is the log in IdentityServer.

**Comment by eaba** [comment]
WWW-Authenticate is not included in the response header.
Like I said The Third layer is not even communicating with identity server to validate the token.
I cannot see any event on Identity server that is triggered by the third layer.

Maybe if you could create sample for my scenario, it would be appreciated.

Thanks!

**Comment by brockallen** [comment]
Sorry, but we have full time jobs and are here only to help when we can. Plus we already have lots of samples. I'd suggest you start with those.

**Comment by eaba** [comment]
I appreciate the time you squeeze out to help.
Please what can be said about the non-inclusion of www-authenticate in the response header - one last time. Thanks

**Comment by eaba** [comment]
Okay I got it working Now.
Though I had been looking at the SampleAspNetWebApi sample, I never studied it.
The key line for me was this: app.UseWebApi(WebApiConfig.Register()); Or app.UseWebApi(Config);
Depending on how your are hosting it. 
Where this is also called is important.
I was calling this from the Global.asax Application_start, that was the reason the response header never had www-authenticate included. So I moved it to the Startup Class and Bingo!! Am all smiles..

Everything is working fine now.

Thanks once again for the time and the great job.
